### PR TITLE
Address comments about metadata docs

### PR DESCRIPTION
**Pull request by hyanwong**
As per initial "minor style suggestion" at https://github.com/tskit-dev/tutorials/pull/143#issuecomment-974149092n - does this read better @LukeAndersonTrocme ?

### CODE REVIEW

File: docs/metadata.md [hunk 1]
```diff
@@ -18,43 +18,45 @@ kernelspec:
...
+metadata automatically and, most importantly, tells downstream users and tools how to
+decode and interpret the metadata. For example, the `msprime` schema for populations
+requires both a `name` and a `description` for each defined population: these names and
+descriptions can assist downstream users in understanding and using `msprime` tree
+sequences. It is best practice to populate such metadata fields if your files will be
+used by any third party, or if you wish to remember what the rows refer to some time
+after making the file!
+
+Technically, schemas describe what information is stored in each metadata record, and
+how it is to be encoded. Every node's metadata follows the node schema, every mutation's
```
Comment (jeromekelleher) [inline docs/metadata.md:40]: maybe also,

> plus some optional rules about the types and ranges of data that can be stored

**Review by jeromekelleher — APPROVED** [review]
LGTM, minor suggestion

**Comment by LukeAndersonTrocme** [comment]
Great update! I like the way Schemas are introduced right from the start, and all those hyperlinks are useful as well. 

Just to nitpick, when you say "Typically these schemas are defined by the software which created the tree-sequence file" I wonder if you could mention or point to an example?

**Comment by hyanwong** [comment]
> Great update! I like the way Schemas are introduced right from the start, and all those hyperlinks are useful as well.
> 
> Just to nitpick, when you say "Typically these schemas are defined by the software which created the tree-sequence file" I wonder if you could mention or point to an example?

I'm not sure many packages do this yet (although there is current discussion on how SLiM is doing it). However, we have mentioned the `msprime` example in the paragraph above, so we could say "typically, as in the example of `msprime` above, schemas are defined by the software which created the tree-sequence file"?

**Comment by LukeAndersonTrocme** [comment]
"typically, as in the example of msprime above, schemas are defined by the software which created the tree-sequence file" is perfect. I asked mainly because I wasn't sure how many other packages supported this already!

**Comment by benjeffery** [comment]
> "typically, as in the example of msprime above, schemas are defined by the software which created the tree-sequence file" is perfect. I asked mainly because I wasn't sure how many other packages supported this already!

We're in very early days of tree sequence metadata, so tools are still integrating it. Thankyou so much for all your feedback, it's invaluable.

**Comment by hyanwong** [comment]
At the moment, "ideally" might be a better adjective than "typically" :)

**Comment by jeromekelleher** [comment]
Are we good to merge here?

**Pull request merged**